packages/payment-proxy-client/src/App.tsx

```diff
@@ -359,20 +359,28 @@ export default function App() {
                       />
                       <FormControl>
                         <RadioGroup
-                          aria-labelledby="demo-radio-buttons-group-label"
                           name="availableFiles"
                           row={true}
-                          value={selectedFileUrl}
+                          value={selectedFile.url}
                           onChange={(e) => {
-                            setSelectedFileUrl(e.target.value);
+                            const found = files.find(
+                              (f) => f.url == e.target.value
+                            );
+                            if (found) {
+                              setSelectedFile(found);
+                            }
                           }}
                         >
                           {files.map((file) => (
                             <FormControlLabel
                               value={file.url}
                               key={file.url}
                               control={<Radio />}
-                              label={file.fileName}
+                              label={
+                                file.fileName.length < 50
+                                  ? file.fileName
+                                  : "..." + file.fileName.slice(-50)
```

Comment:
This trims the filename in case of long file names like we'll have (IE: `DALL·E 2023-09-08 15.21.55 - surrealist painting of weighing scales balanced with one very large apple and tiny small apple.png`)

Reply:
Alternatively we could also introduce an env var for file names, so we can use some shorter names.